src/core/device.c

```diff
@@ -349,6 +361,29 @@ static int device_setup_unit(Manager *m, struct udev_device *dev, const char *pa
                         (void) device_add_udev_wants(u, dev);
         }
 
+        DEVICE(u)->bind_mounts = device_is_bound_by_mounts(dev);
```

Comment:
The reason is that libudev is no more available when device_shall_be_bound_by() is used... I think.